src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/TraceReporter.kt

```diff
@@ -489,17 +506,137 @@ internal class ActorResultNode(
         verboseTrace: Boolean
     ): TraceNode? {
         if (resultRepresentation != null)
-            traceRepresentation.add(TraceEventRepresentation(iThread, traceIndentation() + "result: $resultRepresentation"))
+            traceRepresentation.add(TraceEventRepresentation(iThread, prefix + "result: $resultRepresentation"))
         return next
     }
 }
 
-private const val TRACE_INDENTATION = "  "
+/**
+ * Provides the prefix output for the [TraceNode].
+ * @see TraceNodePrefixFactory
+ */
+internal fun interface PrefixProvider {
```

Comment:
Isn't it always the same string for each instance?

Reply:
What do you mean?

Reply:
It is not the same string, prefix reflects depth call and sometimes includes arrows parts of the spin cycle. See the implementation and usages.

Reply:
When you create a new PrefixProvider, does it always generate the same prefix? Or how does it work? What does `Provider` mean?

Reply:
Please take a look at the `org.jetbrains.kotlinx.lincheck.strategy.managed.TraceNodePrefixFactory` documentation: it is described there.